udaykalra/walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -21,17 +21,38 @@
 import java.util.Set;
 
 /**
- * Returns a collection of TimeRange objects to indicate available meeting times
- * for indicated people and duration given through a request object.
+ * Contains methods for querying availability between attendees.
  */
 public final class FindMeetingQuery {
   private static final int OVERLAP_WITH_SAME_START = 1;
   private static final int OVERLAP_WITH_SAME_END = 2;
   private static final int OVERLAPS_END = 3;
   private static final int OVERLAPS_START = 4;
   private static final int OVERLAP_CONTAINS = 5;
+  private static final int NO_OVERLAP = 0;
 
+  /**
+   * Returns a collection of TimeRange objects to indicate available meeting times
+   * for indicated people and duration given through a request object.
+   */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+    // Perform queries with and without optional attendees.
+    Collection<TimeRange> mandatoryCheck = coreQuery(events, request, false);
+    Collection<TimeRange> optionalCheck = coreQuery(events, request, true);
```

Comment:
Any ideas on clearer names for these variables? Reading these outside of the declaration, I would have a hard time figure out what type these are.

Reply:
Done